src/extensions/askUserForParam.ts

```diff
@@ -157,23 +160,33 @@ async function promptReconfigureSecret(
       } else {
         secretName = await generateSecretName(projectId, instanceId, paramSpec.param);
       }
-
       const secretValue = await promptOnce({
         name: paramSpec.param,
         type: "password",
         message: `This secret will be stored in Cloud Secret Manager as ${secretName}.\nEnter new value for ${paramSpec.label.trim()}:`,
       });
-      if (!secret) {
-        secret = await secretManagerApi.createSecret(
-          projectId,
-          secretName,
-          secretsUtils.getSecretLabels(instanceId)
-        );
+      if (secretValue === "" && paramSpec.required) {
+        logger.info(`Secret value cannot be empty for required param ${paramSpec.param}`);
+        return promptReconfigureSecret(projectId, instanceId, paramSpec);
+      } else if (secretValue !== "") {
+        if (checkResponse(secretValue, paramSpec)) {
+          if (!secret) {
+            secret = await secretManagerApi.createSecret(
+              projectId,
+              secretName,
+              secretsUtils.getSecretLabels(instanceId)
+            );
+          }
+          return addNewSecretVersion(projectId, instanceId, secret, paramSpec, secretValue);
+        } else {
+          return promptReconfigureSecret(projectId, instanceId, paramSpec);
+        }
+      } else {
+        return "";
       }
-      return addNewSecretVersion(projectId, instanceId, secret, paramSpec, secretValue);
     case SecretUpdateAction.LEAVE:
     default:
-      return paramSpec.default || "";
```

Comment:
I believe this should be `paramSpec.default || ""` -- `default` contains previous value, which we want in the `LEAVE` case.

Reply:
Thanks! Updated.